libpromises/rlist.c

```diff
@@ -1044,6 +1059,49 @@ Rlist *RlistFromSplitString(const char *string, char sep)
     return liststart;
 }
 
+/**
+ * Splits the given string into lines. On Windows, both \n and \r\n newlines are
+ * detected. Escaped newlines are respected/ignored too.
+ *
+ * @param detect_crlf whether to try to detect and respect "\r\n" line endings
+ * @return: an #Rlist where items are the individual lines **without** the
+ *          trailing newline character(s)
+ * @note: Free the result with RlistDestroy()
+ * @warning: This function doesn't work properly if @string uses "\r\n" newlines
+ *           and contains '\r' characters that are not part of any "\r\n"
+ *           sequence because it first splits @string on '\r'.
+ */
+Rlist *RlistFromStringSplitLines(const char *string, bool detect_crlf)
+{
+    if (!detect_crlf || (strstr(string, "\r\n") == NULL))
```

Comment:
`strstr` should never be called with `NULL` as one of the arguments.

You already have 

```
if (string == NULL)
```

further down. I suggest moving this up to the top, or adding asserts on top instead.

Reply:
Another good catch.